keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,303 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv import core
+from keras_cv.layers import preprocessing as cv_preprocessing
+from keras_cv.layers.preprocessing.random_augmentation_pipeline import (
+    RandomAugmentationPipeline,
+)
+from keras_cv.utils import preprocessing as preprocessing_utils
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RandAugment(RandomAugmentationPipeline):
+    """RandAugment performs the Rand Augment operation on input images.
+
+    This layer can be thought of as an all in one image augmentation layer.  The policy
+    implemented by this layer has been benchmarked extensively and is effective on a
+    wide variety of datasets.
+
+    The policy operates as follows:
+
+    For each `layer`, the policy selects a random operation from a list of operations.
+    It then samples a random number and if that number is less than
+    `rate` applies it to the given image.
+
+    References:
+        - [RandAugment](https://arxiv.org/abs/1909.13719)
+
+    Args:
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.
+        augmentations_per_image: the number of layers to use in the rand augment policy.
+            Defaults to 3.
+        magnitude: the magnitude to use for each of the augmentation.  magnitude should
```

Comment:
We should clarify in the description that the actual magnitude is random and that this is the mean (before truncation to [0, 1])

Reply:
Updated docstring